### PR TITLE
Bud 104 favoriting topics saves

### LINKED ISSUE #44: Listener removes itself too soon
In ProfileVC, the data accessor listener is initialized in `viewDidLoad` and removed on `viewWillDisappear`, meaning that it stops listening for updates the first time the user navigates away from the tab.

### PR DESCRIPTION
**Pull request by lukemeierj**
Changes in this commit:
* Favoriting a topic in the Topic Tab persists throughout the app and to Firebase
* Changes to Firebase favorited topics appear on the app
* Fixes #44 and #45

### CODE REVIEW

File: Buddies/Topics/TopicsVC.swift [hunk 1]
```diff
@@ -8,13 +8,17 @@
 
 import UIKit
 import AVFoundation
+import FirebaseAuth
```
Comment (noahtallen) [inline Buddies/Topics/TopicsVC.swift]: Do we need to import Auth?

Comment (noahtallen) [inline Buddies/Topics/TopicsVC.swift:11]: Do we need to import Auth?

File: Buddies/Topics/TopicsVC.swift [hunk 2]
```diff
@@ -25,6 +29,7 @@ class TopicsVC: UICollectionViewController, TopicCollectionDelegate {
         } else {
             selectedTopics = selectedTopics.filter({ $0.id != topic.id })
         }
+        user?.favoriteTopics = selectedTopics.map { $0.id }
```
Comment (noahtallen) [inline Buddies/Topics/TopicsVC.swift:32]: Is this what triggers the update with Firebase?

**Review by noahtallen — APPROVED** [review]
LGTM! Seems to work in the simulator ;)

File: Buddies/Topics/TopicsVC.swift (hunk 1 above)
Comment (lukemeierj) [reply Buddies/Topics/TopicsVC.swift:11]: We use it for `func loadProfileData(uid: String = Auth.auth().currentUser!.uid, dataAccess: DataAccessor = DataAccessor.instance) -> Canceler {`

File: Buddies/Topics/TopicsVC.swift (hunk 2 above)
Comment (lukemeierj) [reply Buddies/Topics/TopicsVC.swift:32]: Yep, Jake did a killer job.  Modifying the object updates Firestore.

**Pull request merged**